tools/profiler/src/gemm_operation_profiler.cu

```diff
@@ -373,6 +396,52 @@ void GemmOperationProfiler::initialize_result_(
 
 }
 
+/// Initialize redution problem dimentions and library::Operation
+bool GemmOperationProfiler::initialize_reduction_configuration_(
+  library::Operation const *operation,
+  ProblemSpace::Problem const &problem) {
+
+  library::GemmDescription const &gemm_desc =
+    static_cast<library::GemmDescription const&>(operation->description());
+
+  if (!cast_from_double(problem_.alpha_one, gemm_desc.element_epilogue, 1)) {
+    return false;
+  }
+
+  if (!cast_from_double(problem_.beta_zero, gemm_desc.element_epilogue, 0)) {
+    return false;
+  }
+
+  /// initialize library::ReductionConfiguration
+  gemm_workspace_.reduction_configuration.problem_size      = gemm::GemmCoord(int(problem_.m), int(problem_.n), int(problem_.k)).mn();
+  gemm_workspace_.reduction_configuration.partitions        = int(problem_.split_k_slices);
+  gemm_workspace_.reduction_configuration.partition_stride  = gemm::GemmCoord(int(problem_.m), int(problem_.n), int(problem_.k)).mn().product();
+  gemm_workspace_.reduction_configuration.ldw               = problem_.ldc;
+  gemm_workspace_.reduction_configuration.lds               = problem_.ldc;
+  gemm_workspace_.reduction_configuration.ldd               = problem_.ldc;
+
+  // find reduction operation
+  library::ReductionFunctionalKey reduction_key(
+    library::Provider::kCUTLASS,
+    gemm_desc.tile_description.math_instruction.element_accumulator,    // element workspace
+    gemm_desc.tile_description.math_instruction.element_accumulator,    // element accumulator
+    gemm_desc.C.element,                                                // element output
+    gemm_desc.element_epilogue                                          // element coumpute
+  );
+
+  auto reduction_it = library::Singleton::get().operation_table.reduction_operations.find(reduction_key);
```

Comment:
This should get you the largest aligned reduction kernel that satisfies your problem dimension. At the moment, you only have algin1 kernels though. master has only the largest aligned kernels. Try to make it work for the align4 (F32) reduction case first.

Reply:
@Peter9606 , I am working on merging this PR now.  Sorry that it takes so long.  Would you please elaborate in detail a little bit here how you code can find the max aligned reduction kernel?

Reply:
If I remembered correctly, there isn't a find "procedure" in this PR, as you can see (from this [file](https://github.com/NVIDIA/cutlass/pull/277/files#diff-88b162874290c7ac5efe445498b47a061aeb55905ee7b7ad1797346fc9b5693e) , only one candidate reduction kernel is appended for each alignment.